go/app/infra.go

```diff
@@ -36,8 +43,40 @@ func NewItemRepository() ItemRepository {
 // Insert inserts an item into the repository.
 func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
 	// STEP 4-1: add an implementation to store an item
+	// 既存のデータを読み込む
+	var items []Item
+	filename := "items.json"
+	//file, err := os.Open("items.json")
+	file, err := os.OpenFile(filename, os.O_RDWR|os.O_CREATE, 0664)
```

Comment:
[imo]
`item.json` は `i.fileName` で参照できるので、ここで宣言する必要はなさそうです

```suggestion
	file, err := os.OpenFile(i.fileName, os.O_RDWR|os.O_CREATE, 0664)
```